src/vs/workbench/services/suggest/browser/simpleCompletionModel.ts

```diff
@@ -187,6 +187,16 @@ export class SimpleCompletionModel {
 					return score;
 				}
 			}
+
+			// Check for exact matches, including ignoring file extensions
+			const exactMatchA = a.labelLow === wordLow || (a.completion.isFile && a.labelLowExcludeFileExt === wordLow);
+			const exactMatchB = b.labelLow === wordLow || (b.completion.isFile && b.labelLowExcludeFileExt === wordLow);
+			if (exactMatchA && !exactMatchB) {
+				return -1;
+			} else if (!exactMatchA && exactMatchB) {
+				return 1;
+			}
```

Comment:
Won't this logic clobber the scoring below? The extension stuff was handled down there, do we need to move the scoring down instead?

Reply:
it will only clobber that if there's an exact match, which I think we want?

Reply:
Why doesn't the existing logic around file extensions get hit in this case?

https://github.com/microsoft/vscode/blob/8081bc3f04fb46dcf8755937107ecfc31f6ed96e/src/vs/workbench/services/suggest/browser/simpleCompletionModel.ts#L195-L214

Reply:
bc in the case where a is an exact match and b is not or vice versa, none of the other sorting matters I don't think?